### PR TITLE
[virsh.reboot]Fix the checkpoint of virsh reboot cases

### PR DESCRIPTION
**Pull request by iccaszhulili**
1. Added the process of checking guest boot up time to make
sure the guest rebooted successfully
2. remove the code of sleeping for 5s to wait for the guest
to boot up, since the reboot time for guest is not the same
every time.

Signed-off-by: Lily Zhu <lizhu@redhat.com>

### CODE REVIEW

**Comment by iccaszhulili** [comment]
JOB ID     : 68edfc6403d94c907d382d954e1c2b942bfd2a11
JOB LOG    : /root/avocado/job-results/job-2019-04-21T23.38-68edfc6/job.log
 (1/5) type_specific.io-github-autotest-libvirt.virsh.reboot.normal_test.id_option.no_mode: PASS (79.27 s)
 (2/5) type_specific.io-github-autotest-libvirt.virsh.reboot.normal_test.name_option.no_mode: PASS (67.13 s)
 (3/5) type_specific.io-github-autotest-libvirt.virsh.reboot.normal_test.paused_option.no_mode: PASS (66.91 s)
 (4/5) type_specific.io-github-autotest-libvirt.virsh.reboot.normal_test.uuid_option.no_mode: PASS (64.15 s)

**Comment by yanqzhan** [comment]
LGTM.

File: libvirt/tests/src/virsh_cmd/domain/virsh_reboot.py [hunk 1]
```diff
@@ -90,6 +97,13 @@ def run(test, params, env):
                 logging.error("Exception: %s", str(e))
                 status = -1
         if vm_ref != "remote_name":
+            if not status_error:
+                # Not need to check the boot up time if it is a negative test
+                session = vm.wait_for_login()
+                first_boot_time = session.cmd_output("uptime --since")
+                logging.debug("first_boot_time is %s", first_boot_time)
+                session.close()
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_reboot.py]: I think those are duplicated code? Can we merge them together?

Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_reboot.py]: The function could just return a 'boot_time'. The 'wait_for' below could take a lambda: xxx > first_xx as target function. I think that should do the trick.

Comment (iccaszhulili) [reply libvirt/tests/src/virsh_cmd/domain/virsh_reboot.py]: updated

**Pull request closed without merging**